src/components/input/Button.tsx

```diff
@@ -41,7 +41,7 @@ type ComponentProps = {
 };
 
 export type HTMLButtonAttributes = Omit<
-  JSX.HTMLAttributes<HTMLButtonElement>,
+  JSX.ButtonHTMLAttributes<HTMLButtonElement>,
```

Comment:
Just have a question. Are the new  `JSX.ButtonHTMLAttributes`, `JSX.InputHTMLAttributes`, etc., available in preact versions older than 10.25?

If not, we may need to bump the minimum required preact version of this library, since we ship types with it.

Reply:
Great question: they've were available from `preact/compat` since 10.23 but are only available from `preact` since 10.25.

I see only now there was a peer dependency on preact, sorry! In that case, indeed, you'll probably need a bump.

---

You might be able to work out a system if you need to keep compat with older versions, though it might require some tooling. Essentially, the `HTMLAttributes` used here was a mega interface of all possible attributes (across all elements) which has been renamed to `AllHTMLAttributes` in v10.25+. `HTMLAttributes` in v10.25+ therefore is the purely global attributes/props, things like `class`, `id`, etc. You could probably set up some sort of fallback situation there if you needed.

Reply:
Thanks! Bumping the minimum required version should be fine 🙂